contracts/factory/src/contract.rs

```diff
@@ -34,10 +36,16 @@ pub trait FactoryTrait {
 #[contractimpl]
 impl FactoryTrait for Factory {
     fn initialize(env: Env, admin: Address) {
+        if get_initialized_status(&env) {
+            panic!("Factory: Initialize: initializing contract twice is not allowed");
+        }
+
         save_admin(&env, admin.clone());
 
         save_lp_vec(&env, Vec::new(&env));
 
+        set_initialized_status(&env);
```

Comment:
```suggestion
        if is_initialized(&env) {
            panic!("Factory: Initialize: initializing contract twice is not allowed");
        }
        set_initialized(&env);

        save_admin(&env, admin.clone());

        save_lp_vec(&env, Vec::new(&env));
```

nit: Simplify the names and put the load/save operations together. Easier to catch in case it's missing.